### PR TITLE
Avoid missing pipeline log flood

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: libraries/render/src/render/ShapePipeline.cpp [hunk 1]
```diff
@@ -93,7 +93,11 @@ const ShapePipelinePointer ShapePlumber::pickPipeline(RenderArgs* args, const Ke
 
     const auto& pipelineIterator = _pipelineMap.find(key);
     if (pipelineIterator == _pipelineMap.end()) {
-        qDebug() << "Couldn't find a pipeline from ShapeKey ?" << key;
+        // The first time we can't find a pipeline, we should log it
```
Comment (zzmp) [inline libraries/render/src/render/ShapePipeline.cpp]: I would like to report it as well (to metaverse.highfidelity.com), but the `render` library is not dependent on `networking`, where the `UserActivityLogger` is declared.

This *could* be done with an `invokeMethod` to `qApp`, but that seems sloppy and like technical debt for the future. @ZappoMan do you know if there are other cases where we need to do something similar, and how we got around this?

**Comment by zzmp** [comment]
See: https://app.asana.com/0/74051501495175/118038028990848.

#### Testing

Load a model with a missing pipeline (like http://metaversecafes.com/images/version_d_v7_CLOTHED.fst, but only after #7779). It will flood the log. This limits it to one message per missed pipeline type.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b1b96bb6ce4e8e9fb1896cb4840f91675beb4cb8 available at: https://gist.github.com/85bf2021860c0e6c0311e6f2a78eb267<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7781.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7781.dmg

**Comment by jherico** [comment]
This may fix the pipeline issues, but there's a similar issue with procedural shaders, where we attempt to recompile the shader on every single pass if there's an error.

**Comment by zzmp** [comment]
That's deep in the GLBackend, though, and I don't think it stores state the same way (so you can't stored failed keys in a set, like this).

**Comment by kevinmthomas-carpool** [comment]
testing

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7781
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost
**Additional Test Results/Observations:** Attempted to load ```http://metaversecafes.com/images/version_d_v7_CLOTHED.fst``` and upon loading the avatar it was visible for 2 seconds and crashed with a bugsplat.  Log file attached for additional results.
**Successful Logout/Crash:** Crash.
[hifi-log.txt](https://github.com/highfidelity/hifi/files/245858/hifi-log.txt)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0bc89d6b18474a1d6514ecc2b2b42105da81af41 available at: https://gist.github.com/85bf2021860c0e6c0311e6f2a78eb267<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7781.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7781.dmg

**Comment by kevinmthomas-carpool** [comment]
re-testing

**Comment by kevinmthomas-carpool** [comment]
## QA Re-Test
**PR Build:** 7781
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost
**Additional Test Results/Observations:** Attempted to load ```http://metaversecafes.com/images/version_d_v7_CLOTHED.fst``` and upon loading the avatar it immediately thrusted upward and landed the avatar on the roof of the building in ```localhost```.  The avatar immediately loaded however the loading progress bar displayed that content has not loaded as of yet.  The avatar also had unexpected physics such as the eyes following out of the head tied to a facial joint that was stretched.  The Interface successfully closed and did not crash.  Screen shots and log file attached for additional results.
**Successful Logout/Crash:** Crash.
localhost 1
localhost 2
localhost 3
[hifi-log.txt](https://github.com/highfidelity/hifi/files/247359/hifi-log.txt)

**Comment by zzmp** [comment]
The log flooding was preventing, according to the QA report.

**Comment by zzmp** [comment]
@kevinmthomas-carpool I could not find a BugSplat report for the crash on logout with your latest test. Any chance you've got a mini-dump? If you retest again, does it continue to crash? I'm trying to find out if the crash is related to this PR or not.

**Comment by kevinmthomas-carpool** [comment]
@zzmp was a typo.  We had a successful logout.

**Comment by kevinmthomas-carpool** [comment]
I was trying to say the first work on the PR crashed in the prior build and now it successfully logged out.

**Pull request merged**